spec.emu

```diff
@@ -342,9 +343,11 @@ contributors: Ujjwal Sharma, Younies Mahmoud
                   1. If _durationFormat_.[[MillisecondsStyle]] is *"numeric"*, then
                     1. Set _nextValue_ to _nextValue_ + _duration_.[[Milliseconds]] / 10<sup>3</sup> + _duration_.[[Microseconds]] / 10<sup>6</sup> + _duration_.[[Nanoseconds]] / 10<sup>9</sup>.
                 1. If _nextValue_ is not 0 or _nextDisplay_ is not *"auto"*, then
-                  1. Let _separator_ be _dataLocaleData_.[[digitalFormat]].[[separator]].
+                  1. Let _numberingSystem_ be _durationFormat_.[[NumberingSystem]].
+                  1. Let _separator_ be _dataLocaleData_.[[digitalFormat]].[[&lt;_numberingSystem_&gt;]].
                   1. Append the new Record { [[Type]]: *"literal"*, [[Value]]: _separator_} to the end of _last_.
```

Comment:
This is from the other commit from #133

Reply:
I'm still having a hard time with this, but I think I now understand enough to make a suggestion.
```
        1. Let _numberingSystem_ be _durationFormat_.[[NumberingSystem]].
        1. Let _dataLocale_ be _durationFormat_.[[DataLocale]].
        1. Let _dataLocaleData_ be %DurationFormat%.[[LocaleData]].[[&lt;_dataLocale_&gt;]].
        1. Let _compositeSeparator_ be _dataLocaleData_.[[digitalFormat]].[[&lt;_numberingSystem_&gt;]].
        1. Let _lastStyle_ be *null*.
        1. Let _done_ be *false*.
        1. While _done_ is *false*, repeat for each row in <emu-xref href="#table-duration-components"></emu-xref> in order, except the header row:
          …
          1. Let _valueWithFractionalPart_ be ~empty~.
          1. If _unit_ is *"seconds"* and _durationFormat_.[[MillisecondsStyle]] is *"numeric"*, then
            1. Set _valueWithFractionalPart_ to _value_ + _duration_.[[Milliseconds]] / 10<sup>3</sup> + _duration_.[[Microseconds]] / 10<sup>6</sup> + _duration_.[[Nanoseconds]] / 10<sup>9</sup>.
          1. Else if _unit_ is *"milliseconds"* and _durationFormat_.[[MicrosecondsStyle]] is *"numeric"*, then
            1. Set _valueWithFractionalPart_ to _value_ + _duration_.[[Microseconds]] / 10<sup>3</sup> + _duration_.[[Nanoseconds]] / 10<sup>6</sup>.
          1. Else if _unit_ is *"microseconds"* and _durationFormat_.[[NanosecondsStyle]] is *"numeric"*, then
            1. Set _valueWithFractionalPart_ to _value_ + _duration_.[[Nanoseconds]] / 10<sup>3</sup>.
          1. If _valueWithFractionalPart_ is not ~empty~, then
            1. Set _value_ to _valueWithFractionalPart_.
            1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"maximumFractionDigits"*, _durationFormat_.[[FractionalDigits]]).
            1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"minimumFractionDigits"*, _durationFormat_.[[FractionalDigits]]).
            1. Set _done_ to *true*.

          1. If _value_ is not 0 or _display_ is not *"auto"*, then
            1. If _unitPlaceholders_ is empty, let _lastUnit_ be ~empty~. Otherwise, let _lastUnit_ be the last element of _unitPlaceholders_.
            1. Let _extendPrevious_ be *false*.
            1. NOTE: This area might need tweaking w.r.t. mixing numeric and non-numeric hours/mintues/seconds.
            1. If _lastUnit_ is *"hours"* and _lastStyle_ is *"2-digit"* or *"numeric"* and _unit_ is *"minutes"*, then
              1. Set _extendPrevious_ to *true*.
            1. If _lastUnit_ is *"minutes"* and _lastStyle_ is *"2-digit"* or *"numeric"* and _unit_ is *"seconds"*, then
              1. Set _extendPrevious_ to *true*.
            1. If _extendPrevious_ is *true*, then
              1. Let _partsForListElement_ be _unitPlaceables_.[[&lt;_lastUnit_&gt;]]
              1. Append the new Record { [[Type]]: *"literal"*, [[Value]]: _compositeSeparator_ } to the end of _partsForListElement_.
              1. Set _style_ to *"numeric"*.
            1. Else,
              1. Append _unit_ to _unitPlaceholders_.
              1. Set _unitPlaceables_.[[&lt;_unit_&gt;]] to a new empty List.
              1. Let _partsForListElement_ be _unitPlaceables_.[[&lt;_unit_&gt;]].

            1. If _style_ is *"2-digit"*, then
              1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"minimumIntegerDigits"*, *2*<sub>𝔽</sub>).
            1. Else if _style_ is not *"numeric"*, then
              1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"style"*, *"unit"*).
              1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"unit"*, _numberFormatUnit_).
              1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"unitDisplay"*, _style_).
            1. Let _nf_ be ! Construct(%NumberFormat%, &laquo; _durationFormat_.[[Locale]], _nfOpts_ &raquo;).

            1. Let _partsForUnit_ be ! PartitionNumberPattern(_nf_, 𝔽(_value_)).
            1. For each Record { [[Type]], [[Value]] } _part_ in _partsForUnit_, do
              1. Append the new Record { [[Type]]: _part_.[[Type]], [[Value]]: _part_.[[Value]], [[Unit]]: _numberFormatUnit_ } to the end of _partsForListElement_.
            1. Set _lastStyle_ to _style_.
        1. Let _lfOpts_ be OrdinaryObjectCreate(*null*).
        …
```